### ISSUE TITLE
Segmentation Fault Issue

### ISSUE THREAD

**Issue by BlackCentipede** [mono/CppSharp#433]
I've been trying to get CppSharp to work on Antergos linux distro, I've followed the instruction explained in CppSharp Getting Started documentation to the letter when I cloned, compiled, and try with both versions, specified versions listed in CppSharp compilation instruction and the latest version from these repos:

https://github.com/llvm-mirror/llvm
https://github.com/llvm-mirror/clang

(Initially, when compiling CppSharp libraries, it worked fine until it attempts to test the libraries and cannot find the libraries in Debug_x32 when it's compiled for Release_x64, but it seems non-issue in itself.)

When making a new C# project, I've essentially make a similar code as shown in CppSharp instruction and referenced the libraries as well as copying the shared object libraries to output after compilation of the project and here the code for this: http://pastebin.com/q92TVDYU

I've run and tried, it threw the segmentation faults no matter what option configurations I've tried.

I've done a full wipe of both llvm and clang repos when changing versions between specified version and master version.

I've tried to generate a binding for various libraries including a simple C++ class sample code and all have failed with segmentation fault. C++ Sample Code is this: http://pastebin.com/mSMRmQ2t

I get this stacktrace when attempting to run the code: http://pastebin.com/MJJdW5NH
I get further reading of the backtrace with manual GDB which output this: http://pastebin.com/BT6EFswZ

I end up attempting to read up on CompilerInvocation.cpp under clang/lib/Frontend/ starting at line 1828 aka "clang::CompilerInvocation::CreateFromArgs" function and I was not able to determine why I seems to have segmentation fault issue in this...

I need a bit of guidance in this, please and thank you.

**Comment by tritao** [comment]
It's hard to say why it's crashing... make sure your LLVM and Clang libraries are compiled for the same architecture and bitness as CppSharp and Mono. From the stacktrace it seems to crash in `std::swap` which makes no sense to me.. so I would say it's probably some mismatched configuration somewhere...

Actually are you compiling in 32-bits by any chance? Right now we only support for 64-bits for Linux IIRC. If you're using 32-bits that would explain the crash since the bindings for Clang would have mismatching offsets thus crashing when calling into the native code.

**Comment by BlackCentipede** [comment]
I've checked and verified the complied binary for all dependencies and Mono library itself that everything are in 64 bits.

LLVM and Clang in both compiled build directory and installed system directories are all in elf64-x86-64 format and compiled for i386:x86-64 architecture. I also take extra care to ensure that CppSharp shared object libraries are compiled in 64 bits which show to have same output as above. I also checked /usr/lib/libstdc++.so.6 and it seemed to be compiled same as above.
